File: php/Scheme/ListStats.php
<?php

namespace Scheme;

use Scheme\Interval;
use Scheme\Formula\IFormula;
use Scheme\Formula\Interval\IIntervalFormula;

class ListStats {
	
  private $stats;

  public function __construct() {
    $this->stats = [];
  }
   
    
  private function _getStat($statName) {

    if (isset($this->stats[$statName])) {
      return $this->stats[$statName];
    }
      
    return 0;
  }
    
  public function getStat(IFormula $stat) {

    return $this->_getStat($stat->getName());
  }
    
  public function getRoundStat(IFormula $stat, $afterPoint = -1) {
    $p = $afterPoint > -1 ? $afterPoint : 2;

    return round($this->_getStat($stat->getName()), $afterPoint);
  }
    
  public function setStat(IIntervalFormula $stat, Interval $intv, ListScheme $ls) {

    $this->stats[$stat->getName()] = $stat->getResults($intv, $ls);
  }
}  

File: java/src/scheme/ListStats.java
package scheme;

import java.util.HashMap;

import perslib._Math;
import scheme.Interval;
import formula.IFormula;
import formula.interval.IIntervalFormula;


public class ListStats {
	
	private HashMap<String, Double> stats;
   
    public ListStats(SchemeBase scheme) {
        stats = new HashMap<String, Double>();     
    }   
    
    private double getStat(String statName) {
    	try {
        	return stats.get(statName);
        }
        catch (NullPointerException e) {
        	e.printStackTrace();
        }
        return 0;
    }
    
    public double getStat(IFormula stat) {
    	try {
        return getStat(stat.getName());
    	 }
        catch (NullPointerException e) {
        	//e.printStackTrace();
        	;
        }
        return 0;
    }
    
    public double getRoundStat(IFormula stat) {
    	return _Math.round(getStat(stat.getName()), 2);
    }
    public double getRoundStat(IFormula stat, int afterPoint) {
    	return _Math.round(getStat(stat.getName()), afterPoint);
    }
 
    
    public void setStat(IIntervalFormula stat, Interval intv, ListScheme ls) {
    	try {
    		stats.put(stat.getName(), stat.getResults(intv, ls));
    	}
    	catch (Exception e) {
    		e.printStackTrace();
    	}
    }  

}
